depends/libhdfs3/src/client/OutputStreamImpl.cpp

```diff
@@ -278,7 +316,10 @@ void OutputStreamImpl::append(const char * buf, int64_t size) {
 
 void OutputStreamImpl::appendInternal(const char * buf, int64_t size) {
     int64_t todo = size;
+    if (fileStatus.isFileEncrypted()) {
+        buf = cryptoCodec->encode(buf, size).c_str();
```

Comment:
encode() reture a **temporary** string object, so buf point a uncertain memory.
MUST fix it.